src/vos/vos_obj.c

```diff
@@ -311,8 +311,11 @@ key_punch(struct vos_object *obj, daos_epoch_t epoch, daos_epoch_t bound,
 					 VOS_ITER_AKEY);
 	}
 
-	if (rc != 1)
+	if (rc != 1) {
+		/** Otherwise, key_tree_punch will handle it */
+		rc = vos_key_flags_set(vos_cont2umm(obj->obj_cont), krec);
```

Comment:
Should akey being marked when it's a akey punch?

Reply:
It is handled inside of key_tree_punch